src/sentry/sentry_apps/utils/webhooks.py

```diff
@@ -0,0 +1,89 @@
+from enum import StrEnum
+from typing import Final
+
+
+class SentryAppActionType(StrEnum):
+    pass
+
+
+class IssueActionType(SentryAppActionType):
+    ASSIGNED = "assigned"
+    CREATED = "created"
+    IGNORED = "ignored"
+    RESOLVED = "resolved"
+    UNRESOLVED = "unresolved"
+
+
+class ErrorActionType(SentryAppActionType):
+    CREATED = "created"
+
+
+class CommentActionType(SentryAppActionType):
+    CREATED = "created"
+    DELETED = "deleted"
+    UPDATED = "updated"
+
+
+class MetricAlertActionType(SentryAppActionType):
+    CRITICAL = "critical"
+    OPEN = "open"
+    RESOLVED = "resolved"
+    WARNING = "warning"
+
+
+class IssueAlertActionType(SentryAppActionType):
+    TRIGGERED = "triggered"
+
+
+class InstallationActionType(SentryAppActionType):
+    CREATED = "created"
+    DELETED = "deleted"
+
+
+_sentry_app_webhook_mapping: dict[str, list[str]] = {}
+
+
+class SentryAppResourceType(StrEnum):
+
+    @staticmethod
+    def map_sentry_app_webhook_events(
+        resource: str, action_type: type[SentryAppActionType]
+    ) -> list[str]:
+        # Turn resource + action into webhook event e.g issue.created, issue.resolved, etc.
+        webhook_events = [
+            f"{resource}.{action.value}" for action in action_type._member_map_.values()
+        ]
+        return webhook_events
+
+    ISSUE = "issue"
+    ERROR = "error"
+    COMMENT = "comment"
+    INSTALLATION = "installation"
+    METRIC_ALERT = "metric_alert"
+
+    # Represents an issue alert resource
+    EVENT_ALERT = "event_alert"
+
+
+# When a developer selects to receive "<Resource> Webhooks" it really means
+# listening to a list of specific events. This is a mapping of what those
+# specific events are for each resource.
+EVENT_EXPANSION: Final[dict[SentryAppResourceType, list[str]]] = {
+    SentryAppResourceType.ISSUE: SentryAppResourceType.map_sentry_app_webhook_events(
+        SentryAppResourceType.ISSUE.value, IssueActionType
+    ),
+    SentryAppResourceType.ERROR: SentryAppResourceType.map_sentry_app_webhook_events(
+        SentryAppResourceType.ERROR.value, ErrorActionType
+    ),
+    SentryAppResourceType.COMMENT: SentryAppResourceType.map_sentry_app_webhook_events(
+        SentryAppResourceType.COMMENT.value, CommentActionType
+    ),
+}
+# We present Webhook Subscriptions per-resource (Issue, Project, etc.), not
+# per-event-type (issue.created, project.deleted, etc.). These are valid
+# resources a Sentry App may subscribe to.
+VALID_EVENT_RESOURCES = (
+    SentryAppResourceType.ISSUE,
+    SentryAppResourceType.ERROR,
```

Comment:
Nit: Is there any way we can generate these + the event expansion together? These seem tightly coupled.